tilelang/transform/add_bufstore_wrapper.py

```
@@ -1,43 +1,149 @@
+                    if len(fragment_buffers) == 0:
+                        return statement
+
+                    # Validate fragment buffer indices - only index 0 is supported
+                    buffer_indices = collect_buffer_indices(statement)
+                    for buffer, indices in buffer_indices.items():
+                        if buffer.scope() == "local.fragment":
+                            for index in indices:
+                                if isinstance(index, IntImm) and index != 0:
+                                    raise ValueError(
+                                        f"Fragment buffer access with non-zero index [{index}] is not supported. "
+                                        "Only fragment[0] access is allowed.")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)